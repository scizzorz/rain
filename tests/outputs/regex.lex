keyword 'import'
name 're'
newline
keyword 'import'
name 'array'
newline
keyword 'export'
name 'main'
symbol '='
keyword 'func'
symbol '('
symbol ')'
indent
keyword 'let'
name 'match'
symbol '='
name 're'
symbol '.'
name 'match'
symbol '('
string 'ab*ab*'
symbol ','
string 'abbbab'
symbol ')'
newline
name 'print'
symbol '('
name 'match'
symbol '.'
name 'count'
symbol ')'
newline
keyword 'for'
name 'm'
keyword 'in'
name 'array'
symbol '.'
name 'values'
symbol '('
name 'match'
symbol ')'
indent
name 'print'
symbol '('
name 'm'
symbol ')'
newline
dedent
newline
keyword 'let'
name 'mycompiledregex'
symbol '='
name 're'
symbol '.'
name 'compile'
symbol '('
string 'a(a*)'
symbol ')'
newline
keyword 'let'
name 'match'
symbol '='
name 'mycompiledregex'
symbol ':'
name 'match'
symbol '('
string 'aaaa'
symbol ')'
newline
name 'print'
symbol '('
name 'match'
symbol '.'
name 'count'
symbol ')'
newline
keyword 'for'
name 'm'
keyword 'in'
name 'array'
symbol '.'
name 'values'
symbol '('
name 'match'
symbol ')'
indent
name 'print'
symbol '('
name 'm'
symbol ')'
newline
dedent
newline
keyword 'let'
name 'match'
symbol '='
name 're'
symbol '.'
name 'match'
symbol '?'
symbol '('
string '['
symbol ','
string ''
symbol ')'
newline
name 'print'
symbol '('
name 'match'
symbol '.'
name 'msg'
symbol ')'
newline
dedent
newline
EOF
