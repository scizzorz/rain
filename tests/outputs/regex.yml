!program;1
stmts:
- !import;1
  name: re
  rename: null
- !import;1
  name: array
  rename: null
- !assn;2
  export: true
  let: false
  lhs: !name;1
    value: main
  rhs: !func;1
    body: !block;1
      stmts:
      - !assn;2
        export: false
        let: true
        lhs: !name;1
          value: match
        rhs: !call;1
          args:
          - !str;1
            value: ab*ab*
          - !str;1
            value: abbbab
          catch: false
          func: !index;1
            lhs: !name;1
              value: re
            rhs: !str;1
              value: match
      - !call;1
        args:
        - !index;1
          lhs: !name;1
            value: match
          rhs: !str;1
            value: count
        catch: false
        func: !name;1
          value: print
      - !for;1 {}
      - !assn;2
        export: false
        let: true
        lhs: !name;1
          value: mycompiledregex
        rhs: !call;1
          args:
          - !str;1
            value: a(a*)
          catch: false
          func: !index;1
            lhs: !name;1
              value: re
            rhs: !str;1
              value: compile
      - !assn;2
        export: false
        let: true
        lhs: !name;1
          value: match
        rhs: !method;1
          args:
          - !str;1
            value: aaaa
          catch: false
          lhs: !name;1
            value: mycompiledregex
          rhs: !str;1
            value: match
      - !call;1
        args:
        - !index;1
          lhs: !name;1
            value: match
          rhs: !str;1
            value: count
        catch: false
        func: !name;1
          value: print
      - !for;1 {}
      - !assn;2
        export: false
        let: true
        lhs: !name;1
          value: match
        rhs: !call;1
          args:
          - !str;1
            value: '['
          - !str;1
            value: ''
          catch: true
          func: !index;1
            lhs: !name;1
              value: re
            rhs: !str;1
              value: match
      - !call;1
        args:
        - !index;1
          lhs: !name;1
            value: match
          rhs: !str;1
            value: msg
        catch: false
        func: !name;1
          value: print
    params: []
