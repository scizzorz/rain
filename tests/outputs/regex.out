1
abbbab
2
aaaa
aaa
cannot compile pcre regular expression
